projects/igniteui-angular/src/lib/query-builder/query-builder.component.ts

```diff
@@ -768,10 +768,26 @@ export class IgxQueryBuilderComponent implements AfterViewInit, OnDestroy {
      */
     public deleteGroup() {
         const selectedGroup = this.contextualGroup;
-        const parent = selectedGroup.parent;
+        let parent = selectedGroup.parent;
         if (parent) {
-            const index = parent.children.indexOf(selectedGroup);
+            let index = parent.children.indexOf(selectedGroup);
             parent.children.splice(index, 1);
+
+            if (parent.children.length === 0) {
+                let childGroup = parent;
+                parent = parent.parent;
+                while (parent && parent.children.length === 1) {
+                    childGroup = parent;
+                    parent = parent.parent;
+                }
+    
+                if (parent) {
+                    index = parent.children.indexOf(childGroup);
+                    parent.children.splice(index, 1);
+                } else {
+                    this.rootGroup = null;
+                }
+            }
```

Comment:
We have some code repetiotion here that I believe is possible to be made simpler, probably with a `while` construct.